### ISSUE TITLE
Run-AlValidation: Unexpected error while validating app. Error is: altool.exe not found

### ISSUE THREAD

**Issue by xgwsanh** [microsoft/navcontainerhelper#3476]
PLEASE DO NOT INCLUDE ANY PASSWORDS OR TOKENS IN YOUR ISSUE!!!

**Describe the issue**
When i run RUN-ALVALIDATION, an unexpected error is thrown.

**Scripts used to create container and cause the issue**
```
$validationResults = Run-AlValidation `
    -validateCurrent `
    -apps @("C:\Temp\AuslieferungTest\GWS mbH_Sport and Event Membership Management_23.5.12.0.app") `
    -affixes @("GWS") `
    -countries @("de") `
    -installApps @("C:\Temp\AuslieferungTest\GWS mbH_Sport and Event Base Application_23.5.12.533.app", "C:\Temp\AuslieferungTest\GWS mbH_Sport and Event Contract Management_23.5.17.168.app") `
    -previousApps @("C:\Entwicklung\Sport - Membership Management\app\.alpackages\GWS mbH_Sport and Event Membership Management_23.5.11.443.app")
```

**Full output of scripts**
```
BcContainerHelper version 6.0.15
BC.HelperFunctions emits usage statistics telemetry to Microsoft
Running on Windows, PowerShell 5.1.19041.4170
  _____                               _                
 |  __ \                             | |               
 | |__) |_ _ _ __ __ _ _ __ ___   ___| |_ ___ _ __ ___ 
 |  ___/ _` | '__/ _` | '_ ` _ \ / _ \ __/ _ \ '__/ __|
 | |  | (_| | | | (_| | | | | | |  __/ |_  __/ |  \__ \
 |_|   \__,_|_|  \__,_|_| |_| |_|\___|\__\___|_|  |___/

Container name                  bcserver
Credential                      admin/Goxa5988
MemoryLimit                     8G
validateVersion                 
validateCurrent                 True
validateNextMinor               False
validateNextMajor               False
countries                       de
validateCountries               de
affixes                         GWS
supportedCountries              
ObsoleteTagMinAllowedMajorMinor 
vsixFile                        
Install Apps
- C:\Temp\AuslieferungTest\GWS mbH_Sport and Event Base Application_23.5.12.533.app
- C:\Temp\AuslieferungTest\GWS mbH_Sport and Event Contract Management_23.5.17.168.app
Previous Apps
- C:\Entwicklung\Sport - Membership Management\app\.alpackages\GWS mbH_Sport and Event Membership Management_23.5.11.443.app
Apps
- C:\Temp\AuslieferungTest\GWS mbH_Sport and Event Membership Management_23.5.12.0.app

  _____       _ _ _                                          _        _                            
 |  __ \     | | (_)                                        (_)      (_)                           
 | |__) |   _| | |_ _ __   __ _    __ _  ___ _ __   ___ _ __ _  ___   _ _ __ ___   __ _  __ _  ___ 
 |  ___/ | | | | | | '_ \ / _` |  / _` |/ _ \ '_ \ / _ \ '__| |/ __| | | '_ ` _ \ / _` |/ _` |/ _ \
 | |   | |_| | | | | | | | (_| | | (_| |  __/ | | |  __/ |  | | (__  | | | | | | | (_| | (_| |  __/
 |_|    \__,_|_|_|_|_| |_|\__, |  \__, |\___|_| |_|\___|_|  |_|\___| |_|_| |_| |_|\__,_|\__, |\___|
                           __/ |   __/ |                                                 __/ |     
                          |___/   |___/                                                 |___/      

Pulling mcr.microsoft.com/businesscentral:ltsc2019-filesonly

Pulling generic image took 3 seconds
  _____       _                      _                         _   _  __           _       
 |  __ \     | |                    (_)                       | | (_)/ _|         | |      
 | |  | | ___| |_ ___ _ __ _ __ ___  _ _ __   ___    __ _ _ __| |_ _| |_ __ _  ___| |_ ___ 
 | |  | |/ _ \ __/ _ \ '__| '_ ` _ \| | '_ \ / _ \  / _` | '__| __| |  _/ _` |/ __| __/ __|
 | |__| |  __/ |_  __/ |  | | | | | | | | | |  __/ | (_| | |  | |_| | || (_| | (__| |_\__ \
 |_____/ \___|\__\___|_|  |_| |_| |_|_|_| |_|\___|  \__,_|_|   \__|_|_| \__,_|\___|\__|___/
                                                                                           
Found https://bcartifacts-exdbf9fwegejdqak.b02.azurefd.net/sandbox/24.0.16410.18273/de
Using https://bcartifacts-exdbf9fwegejdqak.b02.azurefd.net/sandbox/24.0.16410.18273/de

   _____                _   _                               _        _                 
  / ____|              | | (_)                             | |      (_)                
 | |     _ __ ___  __ _| |_ _ _ __   __ _    ___ ___  _ __ | |_ __ _ _ _ __   ___ _ __ 
 | |    | '__/ _ \/ _` | __| | '_ \ / _` |  / __/ _ \| '_ \| __/ _` | | '_ \ / _ \ '__|
 | |____| | |  __/ (_| | |_| | | | | (_| | | (__ (_) | | | | |_ (_| | | | | |  __/ |   
  \_____|_|  \___|\__,_|\__|_|_| |_|\__, |  \___\___/|_| |_|\__\__,_|_|_| |_|\___|_|   
                                     __/ |                                             
                                    |___/                                              

Creating container for country de
BcContainerHelper is version 6.0.15
BcContainerHelper is not running as administrator
UsePsSession is True
UsePwshForBc24 is True
Host is Microsoft Windows 10 Enterprise - 10.0.19045.4170
Docker Client Version is 20.10.21
Docker Server Version is 20.10.21
Removing Desktop shortcuts
Fetching all docker images
Fetching all docker volumes
Using image mcr.microsoft.com/businesscentral:ltsc2019-filesonly
Creating Container bcserver
Style: sandbox
Multitenant: Yes
Version: 24.0.16410.18273
Platform: 24.0.18256.0
Generic Tag: 1.0.2.18
Container OS Version: 10.0.17763.5576 (ltsc2019)
Host OS Version: 10.0.19045.4170 (22H2)
WARNING: Host OS and Base Image Container OS doesn't match, defaulting to hyperv. If you do not have HyperV installed or you encounter issues, you could try to specify -isolation process
Using hyperv isolation
Using locale de-DE
Disabling the standard eventlog dump to container log every 2 seconds (use -dumpEventLog to enable)
Downloading C:\ProgramData\BcContainerHelper\Extensions\bcserver\Microsoft.VisualStudio.Services.VSIXPackage.vsix
Downloading using WebClient
Additional Parameters:
--env customNavSettings=EnableTaskScheduler=True
Files in C:\ProgramData\BcContainerHelper\Extensions\bcserver\my:
- AdditionalOutput.ps1
- HelperFunctions.ps1
- MainLoop.ps1
- SetupVariables.ps1
- updatehosts.ps1
Creating container bcserver from image mcr.microsoft.com/businesscentral:ltsc2019-filesonly
99213aab2b79fda3023725a597a94cc6a1d371564d7c6b43a91632e65f2e0c55
Waiting for container bcserver to be ready
Using artifactUrl https://bcartifacts-exdbf9fwegejdqak.b02.azurefd.net/sandbox/24.0.16410.18273/de
Using installer from C:\Run\240
Installing Business Central: multitenant=True, installOnly=False, filesOnly=True, includeTestToolkit=False, includeTestLibrariesOnly=False, includeTestFrameworkOnly=False, includePerformanceToolkit=False, 
appArtifactPath=c:\dl\sandbox\24.0.16410.18273\de, platformArtifactPath=c:\dl\sandbox\24.0.16410.18273\platform, databasePath=c:\dl\sandbox\24.0.16410.18273\de\BusinessCentral-DE.bak, licenseFilePath=c:\dl
\sandbox\24.0.16410.18273\de\Cronus.bclicense, rebootContainer=True
Installing from artifacts
Copying Service Tier Files
c:\dl\sandbox\24.0.16410.18273\platform\ServiceTier\Program Files
c:\dl\sandbox\24.0.16410.18273\platform\ServiceTier\System64Folder
Copying Web Client Files
c:\dl\sandbox\24.0.16410.18273\platform\WebClient\Microsoft Dynamics NAV
Copying ModernDev Files
c:\dl\sandbox\24.0.16410.18273\platform
c:\dl\sandbox\24.0.16410.18273\platform\ModernDev\program files\Microsoft Dynamics NAV
Copying additional files
Copying ConfigurationPackages
C:\dl\sandbox\24.0.16410.18273\de\ConfigurationPackages
Copying Test Assemblies
C:\dl\sandbox\24.0.16410.18273\platform\Test Assemblies
Copying Extensions
C:\dl\sandbox\24.0.16410.18273\de\Extensions
Copying Applications
C:\dl\sandbox\24.0.16410.18273\platform\Applications
Copying Applications.DE
C:\dl\sandbox\24.0.16410.18273\de\Applications.DE
Copying dependencies
Importing PowerShell Modules
Skipping restore of Cronus database
Modifying Business Central Service Tier Config File for Docker
Installing SIP crypto provider: 'C:\Windows\System32\NavSip.dll'
Installation took 5 seconds
Installation complete
Ready for connections!

Creating Container user winrm
Reading CustomSettings.config from bcserver
Cleanup old dotnet core assemblies

Creating container took 39 seconds
  _____                   _                                    _____                           _____            
 |  __ \                 (_)                 /\               / ____|                         / ____|           
 | |__) |   _ _ __  _ __  _ _ __   __ _     /  \   _ __  _ __| (___   ___  _   _ _ __ ___ ___| |     ___  _ __  
 |  _  / | | | '_ \| '_ \| | '_ \ / _` |   / /\ \ | '_ \| '_ \\___ \ / _ \| | | | '__/ __/ _ \ |    / _ \| '_ \ 
 | | \ \ |_| | | | | | | | | | | | (_| |  / ____ \| |_) | |_) |___) | (_) | |_| | | | (__  __/ |____ (_) | |_) |
 |_|  \_\__,_|_| |_|_| |_|_|_| |_|\__, | /_/    \_\ .__/| .__/_____/ \___/ \__,_|_|  \___\___|\_____\___/| .__/ 
                                   __/ |          | |   | |                                              | |    
                                  |___/           |_|   |_|                                              |_|    
Copying previous apps to packages folder
Extracting C:\ProgramData\BcContainerHelper\b2928fcb-024d-41b2-ab32-76b03f3542fe\GWS mbH_Sport and Event Membership Management_23.5.11.443.app
GWS mbH_Sport and Event Membership Management = 23.5.11.443
Run-AlCops Telemetry Correlation Id: f8bce571-8a98-4c13-aeec-901cc7759bad
Unexpected error while validating app. Error is: Command C:\ProgramData\BcContainerHelper\alLanguageExtension\14.0.963280\extension\bin\win32\altool.exe not found, you might need to install that command.
bei CmdDo, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\6.0.15\HelperFunctions.ps1: Zeile 160
bei RunAlTool, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\6.0.15\HelperFunctions.ps1: Zeile 1324
bei Run-AlCops, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\6.0.15\AppHandling\Run-AlCops.ps1: Zeile 126
bei <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\6.0.15\AppHandling\Run-AlValidation.ps1: Zeile 475
bei <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\6.0.15\AppHandling\Run-AlValidation.ps1: Zeile 454
bei <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\6.0.15\AppHandling\Run-AlValidation.ps1: Zeile 397
bei <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\6.0.15\AppHandling\Run-AlValidation.ps1: Zeile 361
bei Run-AlValidation, C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\6.0.15\AppHandling\Run-AlValidation.ps1: Zeile 359
bei <ScriptBlock>, <Keine Datei>: Zeile 1

  _____                           _                _____            _        _                 
 |  __ \                         (_)              / ____|          | |      (_)                
 | |__) |___ _ __ ___   _____   ___ _ __   __ _  | |     ___  _ __ | |_ __ _ _ _ __   ___ _ __ 
 |  _  // _ \ '_ ` _ \ / _ \ \ / / | '_ \ / _` | | |    / _ \| '_ \| __/ _` | | '_ \ / _ \ '__|
 | | \ \  __/ | | | | | (_) \ V /| | | | | (_| | | |____ (_) | | | | |_ (_| | | | | |  __/ |   
 |_|  \_\___|_| |_| |_|\___/ \_/ |_|_| |_|\__, |  \_____\___/|_| |_|\__\__,_|_|_| |_|\___|_|   
                                           __/ |                                               
                                          |___/                                                

Removing Session bcserver
Removing container bcserver
Removing entries from hosts
Removing bcserver from container hosts file
Removing bcserver-* from container hosts file
Removing Desktop shortcuts
Removing C:\ProgramData\BcContainerHelper\Extensions\bcserver

Removing container took 31 seconds

AL Validation finished in 96 seconds
 __      __   _ _     _       _   _               _____                _ _       
 \ \    / /  | (_)   | |     | | (_)             |  __ \              | | |      
  \ \  / /_ _| |_  __| | __ _| |_ _  ___  _ __   | |__) |___ ___ _   _| | |_ ___ 
   \ \/ / _` | | |/ _` |/ _` | __| |/ _ \| '_ \  |  _  // _ \ __| | | | | __/ __|
    \  / (_| | | | (_| | (_| | |_| | (_) | | | | | | \ \  __\__ \ |_| | | |_\__ \
     \/ \__,_|_|_|\__,_|\__,_|\__|_|\___/|_| |_| |_|  \_\___|___/\__,_|_|\__|___/

Unexpected error while validating app. Error is: Command C:\ProgramData\BcContainerHelper\alLanguageExtension\14.0.963280\extension\bin\win32\altool.exe not found, you might need to install that command.
  _____                          ___      __   _ _     _       _   _               ______    _ _                
 |  __ \                   /\   | \ \    / /  | (_)   | |     | | (_)             |  ____|  (_) |               
 | |__) |   _ _ __ ______ /  \  | |\ \  / /_ _| |_  __| | __ _| |_ _  ___  _ __   | |__ __ _ _| |_   _ _ __ ___ 
 |  _  / | | | '_ \______/ /\ \ | | \ \/ / _` | | |/ _` |/ _` | __| |/ _ \| '_ \  |  __/ _` | | | | | | '__/ _ \
 | | \ \ |_| | | | |    / ____ \| |  \  / (_| | | | (_| | (_| | |_| | (_) | | | | | | | (_| | | | |_| | | |  __/
 |_|  \_\__,_|_| |_|   /_/    \_\_|   \/ \__,_|_|_|\__,_|\__,_|\__|_|\___/|_| |_| |_|  \__,_|_|_|\__,_|_|  \___|

```

**Comment by freddydk** [comment]
Does this problem still occur?

**Comment by xgwsanh** [comment]
Problem seems to be gone. Maybe a problem with the used AL Language Extension version?

**Comment by ReindertKorver** [comment]
For any people running into this.
I have had this happen to me as well.  I looked inside the folder and there was no altools file. But I also noticed there was a zip file in C:\ProgramData\BcContainerHelper\alLanguageExtension. I reasoned it probably errored halfway through unzipping. So, I tried deleting the latest folder and the zip file. This triggered BCContainerHelper to create a new folder when I ran the validation again. And that fixed it.
